Core/NoOpRunner.Core/ResourcesUriHandler.cs

```diff
@@ -0,0 +1,185 @@
+
+            return new Uri(AppDomain.CurrentDomain.BaseDirectory + @"..\.." +
+                           System.Text.Encoding.UTF8.GetString(data).Substring(1));
+        }
+
+        public static Uri GetLandingAnimationUri()
+        {
+            spritesReader.GetResourceData("landing", out _, out byte[] data);
+
+            return new Uri(AppDomain.CurrentDomain.BaseDirectory + @"..\.." +
+                           System.Text.Encoding.UTF8.GetString(data).Substring(1));
+        }
+
+        public static Uri GetJumpingAnimationUri()
+        {
+            spritesReader.GetResourceData("jump", out _, out byte[] data);
+
+            return new Uri(AppDomain.CurrentDomain.BaseDirectory + @"..\.." +
+                           System.Text.Encoding.UTF8.GetString(data).Substring(1));
+        }
+
+        public static Uri GetBackground(int backgroundNumber = 5)
+        {
+            string backgroundType = "background";
+
+            if (backgroundNumber < 1 || backgroundNumber > 5)
+            {
+                throw new ArgumentException("1-5 (inclusive)");
+            }
+
+            backgroundType += backgroundNumber;
+
+            spritesReader.GetResourceData(backgroundType, out _, out byte[] data);
+
+            return new Uri(AppDomain.CurrentDomain.BaseDirectory + @"..\.." +
+                           System.Text.Encoding.UTF8.GetString(data).Substring(1));
+        }
+
+        public static Uri GetPlayerPowerUp(VisualElementType visual)
+        {
+            byte[] data;
+            switch (visual)
+            {
+                case VisualElementType.DoubleJump:
+                    spritesReader.GetResourceData("PlayerDoubleJumpVisual", out _, out data);
+
+                    return new Uri(AppDomain.CurrentDomain.BaseDirectory + @"..\.." +
+                                   System.Text.Encoding.UTF8.GetString(data).Substring(1));
+
+                case VisualElementType.Invulnerability:
+                    spritesReader.GetResourceData("PlayerInvulnerabilityVisual", out _, out data);
+
+                    return new Uri(AppDomain.CurrentDomain.BaseDirectory + @"..\.." +
+                                   System.Text.Encoding.UTF8.GetString(data).Substring(1));
+                case VisualElementType.SpeedBoost:
+                    spritesReader.GetResourceData("PlayerSpeedBoostVisual", out _, out data);
+
+                    return new Uri(AppDomain.CurrentDomain.BaseDirectory + @"..\.." +
+                                   System.Text.Encoding.UTF8.GetString(data).Substring(1));
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(visual), visual, null);
+            }
+        }
+
+        public static Uri GetPowerUp(PowerUps powerUp)
+        {
+            byte[] data;
+            switch (powerUp)
+            {
+                case PowerUps.Speed_Boost:
+                    spritesReader.GetResourceData("SpeedBoost", out _, out data);
```

Comment:
Why isn't this codeblock just a method that takes the string as param. Now these three lines are copy pasted way too many times.